test/unittest/tests/block/test_block_api_write.cpp

```diff
@@ -0,0 +1,279 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_api]: block_write.c
+ * The block manager performs file operations on specific blocks which eventually gets translated
+ * onto files. This test suite aims to test the possible cases for the write(), read()
+ * and write_size() API operations.
+ */
+#include <catch2/catch.hpp>
+#include <iostream>
+#include <filesystem>
+
+#include "wt_internal.h"
+#include "../wrappers/mock_session.h"
+#include "../wrappers/config_parser.h"
+#include "../wrappers/item_wrapper.h"
+
+const int DEFAULT_BLOCK_SIZE = 256;
+const std::string ALLOCATION_SIZE = "256";
+const std::string BLOCK_ALLOCATION = "best";
+const std::string OS_CACHE_MAX = "0";
+const std::string OS_CACHE_DIRTY_MAX = "0";
+const std::string ACCESS_PATTERN = "random";
+const std::string DEFAULT_FILE_NAME = "test.txt";
+
+typedef std::pair<std::array<uint8_t, WT_BTREE_MAX_ADDR_COOKIE>, size_t> addr_cookie;
+
+void
+test_and_validate_write_size(WT_BM *bm, std::shared_ptr<mock_session> session, size_t size)
+{
+    size_t init_size = size;
+    REQUIRE(bm->write_size(bm, session->get_wt_session_impl(), &size) == 0);
+    CHECK(size % DEFAULT_BLOCK_SIZE == 0);
+    CHECK(size == ((init_size / DEFAULT_BLOCK_SIZE) + 1) * DEFAULT_BLOCK_SIZE);
+}
+
+void
+validate_block_contents(WT_BM *bm, std::shared_ptr<mock_session> session, WT_ITEM *write_buf,
+  addr_cookie cookie, wt_off_t offset, uint32_t size)
+{
+    // Test that the write was correctly written via __wt_read().
```

Comment:
In general this file needs more comments, but in this function specifically can we be more detailed? I also think we should do the block manager read first and then the non block manager read. This makes our test more like black box then white box.

// Test that the write was correctly written via __wt_read(). -> Using the non-block manager read function read the file where the block should've been written. Then compare that with the original write buffer.

// Test that the write buffer was correctly via bm->read() -> Using the block manager read function read in the block using the address cookie. Testing the using the bm->read() function isn't a complete test as it could be prone to the same bugs as the write function. We check this by doing a standard WT read later on.

Reply:
As discussed I clear write_buf checksum is because the read buffer is cleared before returning.